### PR TITLE
CNV-19298: SSH access with virtctl command

### PR DESCRIPTION
**Pull request by upalatucci**
## 📝 Description

SSH access (VM details page section) with virtctl and ssh commands

Screenshot from 2022-08-24 16-43-49
Screenshot from 2022-08-24 16-43-47

### CODE REVIEW

File: src/utils/components/SSHAccess/utils.ts [hunk 1]
```diff
@@ -68,9 +71,21 @@ export const createSSHService = async (vm: V1VirtualMachine, vmi: V1VirtualMachi
...
 };
+
+export const getConsoleVirtctlCommand = (vmName: string, vmNamespace?: string) => {
+  let command = 'virtctl ';
+
+  if (vmNamespace) {
+    command += `-n ${vmNamespace} `;
+  }
+
+  command += vmName;
```
Comment (hstastna) [inline src/utils/components/SSHAccess/utils.ts:88]: Are you sure about the command `virtctl -n <vmNamespace> <vmName>` ? According to the doc, we are missing ` ssh ` there...

**Review by hstastna — COMMENTED** [review]
Just one comment below, otherwise great work! 👍🏽

File: src/utils/components/SSHAccess/utils.ts (hunk 1 above)
Comment (upalatucci) [reply src/utils/components/SSHAccess/utils.ts:88]: Yes! I've installed an old version and there was no ssh command. Now I installed the 0.53 and there is!

File: src/utils/components/SSHAccess/utils.ts [hunk 2]
```diff
@@ -68,9 +71,21 @@ export const createSSHService = async (vm: V1VirtualMachine, vmi: V1VirtualMachi
...
+export const getConsoleVirtctlCommand = (vmName: string, vmNamespace?: string) => {
+  let command = 'virtctl ';
+
+  if (vmNamespace) {
+    command += `-n ${vmNamespace} `;
+  }
+
+  command += `ssh ${vmName}`;
+
+  return command;
```
Comment (metalice) [inline src/utils/components/SSHAccess/utils.ts:90]: `virtctl {$vmNamespace ? -n ${vmNamespace} : ""} ssh ${vmName}` - not sure if this function is needed

File: src/utils/components/SSHAccess/utils.ts [hunk 3]
```diff
@@ -68,9 +71,21 @@ export const createSSHService = async (vm: V1VirtualMachine, vmi: V1VirtualMachi
 
   const serviceResource = buildSSHServiceFromVM(vm, labelSelector);
 
-  await k8sCreate({
+  return await k8sCreate({
```
Comment (metalice) [inline src/utils/components/SSHAccess/utils.ts:74]: return await is same as return of regular promise

File: src/utils/components/SSHAccess/components/SSHCheckbox.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  const [sshService, setSSHService] = React.useState(initialSSHService);
+  const [loading, setLoading] = React.useState(false);
+  const [error, setError] = React.useState<Error>();
+
+  const onSSHChange = React.useCallback(
+    async (enableSSH: boolean) => {
+      setLoading(true);
+      try {
+        if (enableSSH) {
+          setSSHService((await createSSHService(vm, vmi)) as IoK8sApiCoreV1Service);
```
Comment (metalice) [inline src/utils/components/SSHAccess/components/SSHCheckbox.tsx:37]: u are saving a promise.. no point in await here..

Comment (metalice) [reply src/utils/components/SSHAccess/components/SSHCheckbox.tsx:37]: if you arent dealing with the result of await in the same scope it just the same as regular promise

Comment (metalice) [reply src/utils/components/SSHAccess/components/SSHCheckbox.tsx:37]: maybe createahSSHService.then(()=>setSSHService()) ? can work?

Comment (metalice) [reply src/utils/components/SSHAccess/components/SSHCheckbox.tsx:37]: just remove the finally and move it also to .then

File: src/utils/components/SSHAccess/utils.ts (hunk 2 above)
Comment (upalatucci) [reply src/utils/components/SSHAccess/utils.ts:90]: Surely Hilda will use it in another place for the dropdown action so I thought a function would be useful

**Review by hstastna — APPROVED** [review]
LGTM!

**Pull request merged**